doc/man3/SRP_VBASE_new.pod

```diff
@@ -0,0 +1,84 @@
+=pod
+
+=head1 NAME
+
+SRP_VBASE_new,
+SRP_VBASE_free,
+SRP_user_pwd_free,
+SRP_VBASE_init,
+SRP_VBASE_get1_by_user,
+SRP_VBASE_get_by_user
+- Functions to create and manage a stack of SRP user verifier information
+
+=head1 SYNOPSIS
+
+ #include <openssl/srp.h>
+
+ SRP_VBASE *SRP_VBASE_new(char *seed_key);
+ void SRP_VBASE_free(SRP_VBASE *vb);
+ void SRP_user_pwd_free(SRP_user_pwd *user_pwd);
+
+ int SRP_VBASE_init(SRP_VBASE *vb, char *verifier_file);
+
+ SRP_user_pwd *SRP_VBASE_get1_by_user(SRP_VBASE *vb, char *username);
+ SRP_user_pwd *SRP_VBASE_get_by_user(SRP_VBASE *vb, char *username);
+
+=head1 DESCRIPTION
+
+The SRP_VBASE_new() function allocates a structure to store server side SRP
+verifier information. If B<seed_key> is not NULL a copy is stored and used to
+generate dummy parameters for users that are not found by SRP_VBASE_get1_by_user().
+This allows the server to hide the fact that it doesn't have a verifier for a
+particular username, as described in section 2.5.1.3 'Unknown SRP' of RFC 5054.
+The seed string should contain random NULL terminated binary data (therefore
+the random data should not contain NULL bytes!).
+
+The SRP_VBASE_free() function frees up the B<vb> structure.
+If B<vb> is NULL, nothing is done.
+
+The SRP_user_pwd_free() function frees up the B<user_pwd> structure.
+If B<user_pwd> is NULL, nothing is done.
+
+The SRP_VBASE_init() function parses the information in a verifier file and
+populates the B<vb> structure.
+
+The SRP_VBASE_get1_by_user() function returns the password info for the user
+whose username matches B<username>. It replaces the deprecated
+SRP_VBASE_get_by_user().
+If no matching user is found but a seed_key and default gN parameters have been set,
+dummy authentication information is generated from the seed_key. This will cause
+the handshake to proceed normally but the first client will be rejected with
+a "bad_record_mac" alert, as if the password was incorrect.
```

Comment:
This is TLS specific behaviour - but this API is generic, so we need a more general form of words here.